### PR TITLE
Handle error message for unsupported version in swagger for IDL Client 

### PR DESCRIPTION
**Pull request by lnash94**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: openapi-core/src/main/java/io/ballerina/openapi/core/ErrorMessages.java [hunk 1]
```diff
@@ -36,4 +36,9 @@ public static String invalidFileType() {
     public static String invalidPathParamType(String parameterName) {
         return String.format("Invalid path parameter data type for the parameter: '%s'", parameterName);
     }
+
+    public static String unsupportedOpenAPIVersion() {
+        return String.format("given openAPI definition version is not supported via openAPI tool. " +
+                "Please use openAPI definition version 03 or higher");
```
Comment (NipunaRanasinghe) [inline openapi-core/src/main/java/io/ballerina/openapi/core/ErrorMessages.java:42]:
```suggestion
        return String.format("Provided openAPI contract version is not supported in openAPI tool. " +
                "Use openAPI specification version 3 or higher");
```

File: openapi-client-idl-plugin/src/main/java/io/ballerina/openapi/idl/client/OpenAPIClientGenerator.java [hunk 2]
```diff
@@ -151,17 +151,19 @@ public void perform(IDLSourceGeneratorContext idlSourceContext) {
...
                 return false;
             }
             String content = Files.readString(oasPath);
-            Pattern pattern = Pattern.compile("openapi:");
-            Matcher matcher = pattern.matcher(content);
-            return matcher.find();
+            Pattern swaggerPattern = Pattern.compile("swagger:");
+            Pattern openapiPattern = Pattern.compile("openapi:");
+            Matcher openapiMatcher = openapiPattern.matcher(content);
+            Matcher swaggerMatcher = swaggerPattern.matcher(content);
```
Comment (NipunaRanasinghe) [inline openapi-client-idl-plugin/src/main/java/io/ballerina/openapi/idl/client/OpenAPIClientGenerator.java:165]: Shall we improve the openAPI contract detection by including the version regex as well? That will further reduce the chance of false positives.

Comment (lnash94) [reply openapi-client-idl-plugin/src/main/java/io/ballerina/openapi/idl/client/OpenAPIClientGenerator.java:165]: Yeah here we checked the version of YAML. when it is with `swagger` , it should be version 2. When it starts with `openapi`, it version should be 3 or higher.  :)

Comment (NipunaRanasinghe) [reply openapi-client-idl-plugin/src/main/java/io/ballerina/openapi/idl/client/OpenAPIClientGenerator.java:165]: Yes but still we can use a regex expression to capture the entire version definition line, as the version is semver compatible

Comment (NipunaRanasinghe) [reply openapi-client-idl-plugin/src/main/java/io/ballerina/openapi/idl/client/OpenAPIClientGenerator.java:165]: We can even further improve this logic to use a single regex which can capture both the swagger and openAPI cases.

Comment (lnash94) [reply openapi-client-idl-plugin/src/main/java/io/ballerina/openapi/idl/client/OpenAPIClientGenerator.java]: +1 for one regex for addressing both versions. fixed with https://github.com/ballerina-platform/openapi-tools/pull/1086/commits/620687cf94857743cee7d275bfcb4cf0d66a834a

File: openapi-client-idl-plugin/src/main/java/io/ballerina/openapi/idl/client/Constants.java [hunk 3]
```diff
@@ -33,6 +33,8 @@ public class Constants {
     public static final String LICENSE = "license";
     public static final String TRUE = "true";
     public static final String MODULE_ALIAS = "openapi_client";
+    public static final String OPENAPI_REGEX_PATTERN = "(openapi:\\s*\"*[0-9]\\d*\\.\\d+\\" +
+            ".\\d+)|(swagger:\\s*\"*[0-9]\\d*\\.\\d+\\.\\d+)";
```
Comment (NipunaRanasinghe) [inline openapi-client-idl-plugin/src/main/java/io/ballerina/openapi/idl/client/Constants.java:37]:
```suggestion
    public static final String OPENAPI_REGEX_PATTERN = "(openapi|swagger):\\s*\\d+\\.\\d+\\.\\d+";
```

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

**Pull request merged**